offlineimap/imapserver.py

```diff
@@ -198,7 +198,7 @@ def __md5handler(self, response):
         challenge = response.strip()
         self.ui.debug('imap', '__md5handler: got challenge %s'% challenge)
 
-        passwd = self.__getpassword()
+        passwd = self.__getpassword().encode('ascii')
```

Comment:
The u prefix is only usefull if there are non-ASCII characters. I agree the configuration file would need more details about when to use unicode.

There is no "cheap" excuse but I've probably not provided enough background.

1. For now, this is the only patch addressing all the issues raised here in a clean way for both py2 and py3.

2. There was a recent attempt to improve this without success #324 and I'm addressing this, too.

3. This is making UTF-8 support consistent. I've merged patches when good enough the way they came in. However, nobody got the full picture and provided patches for their own use cases only. Currently, the code supports UTF-8 for only some auth mechanisms which is something quite ugly. My patch is addressing this, too. There is only one last thing that I'm not addressing: the netrc feature.